### DISCUSSION TITLE
Proper Error Propagation for Triggers on auth.users Table

### DISCUSSION THREAD

**Discussion by TheCodeDestroyer** [supabase/supabase#28296] [Feature Requests] [answered]
Hey Supabase team! 👋

I’m using Supabase with an OAuth provider in my NextJS app, and I ran into an issue with error handling.
When a trigger function on the auth.users table raises an exception, the error doesn’t make its way back to the `/api/auth/callback endpoint`.
Right now the query param error_description just says `Database error saving new user` and the error_code is `500`,
which makes it really hard to handle edge case flows if I have some custom logic around user creation.

**Comment by GaryAustin1** [comment, marked as answer]
https://github.com/supabase/auth/issues/1602

**Reply by TheCodeDestroyer** [reply]
@GaryAustin1 Thank you for confirming the hooks are the planned solution.
I wasnt quite sure if thats the same since hooks are currently specific to sending sms and email and MFA verification.
No hook for user creation or deletion.
Do you maybe have a link to the issue that has the checklist for planned hooks?

**Reply by GaryAustin1** [reply]
I have not seen any roadmap.
https://github.com/supabase/auth/blob/6ccd814309dca70a9e3585543887194b05d725d3/internal/hooks/auth_hooks.go#L4
Seems to be where they add new hooks.
